src/nvim/buffer.c

```diff
@@ -117,7 +117,7 @@ read_buffer(
       flags | READ_BUFFER);
   if (retval == OK) {
     // Delete the binary lines.
-    while (--line_count >= 0) {
+    while (line_count-- > 0) {
```

Comment:
This is a case we would not change, and may cause a bug. 

pre/post-decrement have different behavior in expressions (as opposed to statements). The linter only warns about about _statements_, not expressions.